src/config_generator/from_proto.rs

```diff
@@ -0,0 +1,461 @@
+        set.file.push(greetings_b);
+
+        let result = from_proto(vec![set], "Query").to_sdl();
+
+        insta::assert_snapshot!(result);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_config_from_sdl() -> anyhow::Result<()> {
+        let mut set = FileDescriptorSet::default();
+
+        let news = get_proto_file_descriptor("news.proto")?;
+        let greetings_a = get_proto_file_descriptor("greetings_a.proto")?;
+        let greetings_b = get_proto_file_descriptor("greetings_b.proto")?;
+
+        set.file.push(news.clone());
+        set.file.push(greetings_a.clone());
+        set.file.push(greetings_b.clone());
+
+        let result = from_proto(vec![set], "Query").to_sdl();
+
+        // test for different sets
+        let mut set = FileDescriptorSet::default();
+        let mut set1 = FileDescriptorSet::default();
+        let mut set2 = FileDescriptorSet::default();
+        set.file.push(news);
+        set1.file.push(greetings_a);
+        set2.file.push(greetings_b);
+
+        let result_sets = from_proto(vec![set, set1, set2], "Query").to_sdl();
+
+        pretty_assertions::assert_eq!(result, result_sets);
+        Ok(())
+    }
+
+    #[test]
+    fn test_required_types() -> anyhow::Result<()> {
+        // required fields are deprecated in proto3 (https://protobuf.dev/programming-guides/dos-donts/#add-required)
+        // this example uses proto2 to test the same.
+        // for proto3 it's guaranteed to have a default value (https://protobuf.dev/programming-guides/proto3/#default)
+        // and our implementation (https://github.com/tailcallhq/tailcall/pull/1537) supports default values by default.
+        // so we do not need to explicitly mark fields as required.
+
+        let mut set = FileDescriptorSet::default();
+        let req_proto = get_proto_file_descriptor("person.proto")?;
+        set.file.push(req_proto);
+
+        let cfg = from_proto(vec![set], "Query").to_sdl();
+        insta::assert_snapshot!(cfg);
+
+        Ok(())
+    }
+    #[test]
+    fn test_get_value_enum() {
+        let ctx: Context = Context::new("Query").package("com.example".to_string());
+        assert_eq!(
+            ctx.get_name("TestEnum", DescriptorType::Enum),
+            "ComExample__TestEnum"
+        );
+    }
+
+    #[test]
+    fn test_get_value_message() {
+        let ctx: Context = Context::new("Query").package("com.example".to_string());
+        assert_eq!(
+            ctx.get_name("testMessage", DescriptorType::Message),
+            "ComExample__testMessage"
+        );
+    }
+
+    #[test]
+    fn test_get_value_query_name() {
+        let ctx: Context = Context::new("Query").package("com.example".to_string());
+        assert_eq!(
+            ctx.get_name("QueryName", DescriptorType::Operation),
+            "ComExample__queryName"
+        );
+    }
+
+    #[test]
+    fn test_insert_and_get_enum() {
+        let mut ctx: Context = Context::new("Query").package("com.example".to_string());
```

Comment:
please split this into individual tests. Use better names for the test.

Follow the guideline here https://github.com/tailcallhq/tailcall/blob/main/.github/contributing.md#testing

Reply:
negative